### DISCUSSION TITLE
Using daisyui colors to theme 'non-tailwind' components

### DISCUSSION THREAD

**Discussion by sbahman** [saadeghi/daisyui#1579] [Q&A] [answered]
I have a `Chart` component which takes a `DataSets` object as a parameter. This object also contains the information about the border and background color of the chart.

```
const [dataSets, setDataSets] = useState<DataSets | null>(null)

  useEffect(() => {
    fetch(dataUrl)
      .then((response) => response.json())
      .then((D) => {
        setDataSets({
          labels: D.date,
          datasets: [
            {
              label: 'Series',
              data: D.values,
              borderColor: main_page_colors.primary,
              backgroundColor: main_page_colors.primary + '0D',
            },
          ],
        })
      })
  }, [])

  return (
    <div className="h-[32rem]">
      {dataSets == null ? (
        <></>
      ) : (
        <Chart data={dataSets} options={options} type="line" />
      )}
    </div>
  )
}
```

I would like to match the theme of this chart with that of DaisyUI. What is the best way to do this? Ideally the charts theme could also be changed when the user changes the DaisyUi theme with `theme-change` but if thats not possible I would at least like to match one theme.

**Comment by saadeghi** [comment, marked as answer]
If you can set CSS variables as color values for the chart, it would be easy. You can use CSS variable of each daisyUI color. For example `--p` or `--s` ([see this table](https://daisyui.com/docs/colors/))

If not, another way would be writing some custom CSS to set daisyUI color values to each section of chart if it's possible to select them using CSS selctors.
